Source/Tests/LinuxSyscalls/FileManagement.cpp

```diff
@@ -293,52 +268,71 @@ FileManager::FileManager(FEXCore::Context::Context *ctx)
         },
       };
 
+      // If a thunks DB property exists then we pull in data from the thunks database
       json_t const *json = json_createWithPool(&FileData.at(0), &Pool.PoolObject);
       json_t const* ThunksDB = json_getProperty( json, "ThunksDB" );
-      if (ThunksDB) {
-        LoadThunksDB(&LoadedThunkDatabase, ThunksDB);
+      if (!ThunksDB) {
+        continue;
+      }
+
+      for (json_t const* Item = json_getChild(ThunksDB); Item != nullptr; Item = json_getSibling(Item)) {
+        const char *LibraryName = json_getName(Item);
+        bool LibraryEnabled = json_getInteger(Item) != 0;
+        // If the library is enabled then find it in the DB
+        auto DBObject = ThunkDB.find(LibraryName);
+        if (DBObject != ThunkDB.end()) {
+          DBObject->second.Enabled = LibraryEnabled;
+        }
       }
     }
   }
 
-  // Now that we loaded the thunks object, walk through and ensure dependencies are enabled as well.
+  // Now that we loaded the thunks object, walk through and ensure dependencies are enabled as well
+  auto ThunkGuestPath = std::filesystem::path { Is64BitMode() ? ThunkGuestLibs() : ThunkGuestLibs32() };
   for (auto const &DBObject : ThunkDB) {
     if (!DBObject.second.Enabled) {
       continue;
     }
 
-    // Now walk the dependencies and set them up as well
-    // Make sure to enable each one as we go to remove circular dependencies
-    std::function<void(const std::unordered_set<std::string> &Depends, bool AlreadyEnabled)> InsertDependencies
-      = [this, &ThunkGuestPath, &InsertDependencies](const std::unordered_set<std::string> &Depends, bool AlreadyEnabled) -> void {
-      for (auto const &Depend : Depends) {
-        auto DBDepend = ThunkDB.find(Depend);
-        if (DBDepend != ThunkDB.end() &&
-            (DBDepend->second.Enabled == false || AlreadyEnabled)) {
+    // Recursively add paths for this thunk library and its dependencies to ThunkOverlays.
+    // Using a local struct for this is slightly less ugly than using self-capturing lambdas
+    struct {
+      decltype(FileManager::ThunkOverlays)& ThunkOverlays;
+      decltype(ThunkDB)& ThunkDB;
+      const std::filesystem::path& ThunkGuestPath;
 
-          auto ThunkPath = ThunkGuestPath / DBDepend->second.LibraryName;
+      void SetupOverlay(const ThunkDBObject& DBDepend) {
+          auto ThunkPath = ThunkGuestPath / DBDepend.LibraryName;
           if (!std::filesystem::exists(ThunkPath)) {
-              ERROR_AND_DIE_FMT("Requested thunking via guest library \"{}\" that does not exist", ThunkPath.string());
+            ERROR_AND_DIE_FMT("Requested thunking via guest library \"{}\" that does not exist", ThunkPath.string());
```

Comment:
Oop, just tested with this and this error needs to be non-fatal.
A common use case is to enable thunks globally while testing, ignoring if any 32-bit games accidentally are run.
So changing this over to a `LogMan::Msg::IFmt` is going to be necessary.

Reply:
Should be resolved now by `return`ing on 32-bit instead of printing an error.